src/Url.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the webmozart/path-util package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Webmozart\PathUtil;
+
+use InvalidArgumentException;
+use Webmozart\Assert\Assert;
+
+/**
+ * Contains utility methods for handling URL strings.
+ *
+ * The methods in this class are able to deal with URLs.
+ *
+ * @since  2.3
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Claudio Zizza <claudio@budgegeria.de>
+ */
+final class Url
+{
+    /**
+     * Turns an URL into a relative path.
+     *
+     * The result is a canonical path. This class is using functionality of Path class.
+     *
+     * @see Path
+     *
+     * @param string $url     An URL to make relative.
+     * @param string $baseUrl A base path or URL.
+     *
+     * @return string
+     *
+     * @throws InvalidArgumentException If the hostname of url and baseUrl does
+     *                                  not match.
+     */
+    public static function makeRelative($url, $baseUrl)
+    {
+        Assert::string($url, 'The path must be a string. Got: %s');
+        Assert::string($baseUrl, 'The base path must be a string. Got: %s');
+
+        list($root, $relativePath) = self::split($url);
+        list($baseRoot, $relativeBasePath) = self::split($baseUrl);
+
+        $path = Path::makeRelative($relativePath, $relativeBasePath);
+
+        if ('' !== $root && '' !== $baseRoot && $root !== $baseRoot) {
+            throw new InvalidArgumentException(sprintf(
+                'Domain "%s" doesn\'t equal to base "%s".',
```

Comment:
The URL "%s" cannot be made relative to "%s" since their host names are different.